front/tests/012-op-simulation-settings-tab.spec.ts

```diff
@@ -0,0 +1,363 @@
+    // TODO: Remove the reload when bug #8854 (UI not updating after modification) is fixed
+    await page.reload();
+    await opTimetablePage.verifyTimeStopsDataSheetVisibility();
+    await opTimetablePage.getTrainArrivalTime('12:05');
+    await opTimetablePage.clickOnScenarioCollapseButton();
+    await opOutputTablePage.getOutputTableData(
+      expectedCellDataElectricalProfileOFF,
+      selectedLanguage
+    );
+  });
+  test('should correctly activate composition code', async ({ page }) => {
+    const [
+      opInputTablePage,
+      opTimetablePage,
+      opOutputTablePage,
+      scenarioPage,
+      opSimulationSettingsPage,
+    ] = [
+      new OperationalStudiesInputTablePage(page),
+      new OperationalStudiesTimetablePage(page),
+      new OperationalStudiesOutputTablePage(page),
+      new ScenarioPage(page),
+      new OperationalStudiesSimulationSettingsPage(page),
+    ];
+
+    const translations = selectedLanguage === 'English' ? enTranslations : frTranslations;
+    const cell: CellData = {
+      stationName: 'Mid_East_station',
+      header: 'stopTime',
+      value: '124',
+    };
+    const translatedHeader = cleanWhitespace(translations[cell.header]);
+
+    await opInputTablePage.fillTableCellByStationAndHeader(
+      cell.stationName,
+      translatedHeader,
+      cell.value,
+      selectedLanguage
+    );
+    // Select a specific composition code option
+    await scenarioPage.openTabByDataId('tab-simulation-settings');
+    await opSimulationSettingsPage.deactivateElectricalProfile();
+    await opSimulationSettingsPage.activateMarecoMargin();
+    await opSimulationSettingsPage.selectCodeCompoOption('HLP');
+    // Add the train schedule and verify output results
+    await scenarioPage.addTrainSchedule();
+    await scenarioPage.returnSimulationResult();
+    // TODO: Remove the reload when bug #8854 (UI not updating after modification) is fixed
+    await page.reload();
+    await opTimetablePage.verifyTimeStopsDataSheetVisibility();
+    await opTimetablePage.getTrainArrivalTime('11:54');
+    await opTimetablePage.clickOnScenarioCollapseButton();
+    await scrollContainer(page, '.osrd-simulation-container .time-stops-datasheet .dsg-container');
+    await opOutputTablePage.getOutputTableData(expectedCellDataCodeCompoON, selectedLanguage);
+    await opTimetablePage.clickOnTimetableCollapseButton();
+    // Modify the composition code option and verify the changes
+    await opTimetablePage.clickOnEditTrain();
+    await scenarioPage.openTabByDataId('tab-simulation-settings');
+    await opSimulationSettingsPage.selectCodeCompoOption('__PLACEHOLDER__');
+    await opTimetablePage.clickOnEditTrainSchedule();
+    // TODO: Remove the reload when bug #8854 (UI not updating after modification) is fixed
+    await page.reload();
+    await opTimetablePage.verifyTimeStopsDataSheetVisibility();
+    await opTimetablePage.getTrainArrivalTime('12:05');
```

Comment:
If I understand correctly, we expect that the train is slower without any code compo than with one ? Is that normal ? (maybe I'm just missing a metier logic)

Reply:
The train is slower with code compo:  it was 11:54 with and 12:05 without => Updated the comment

Reply:
Agreed but if the train is slower with a compo it should arrived later than without no ? But here we expect the "slow" train (compo HLP) to arrive at 11:54 and the "fast" (no compo) to arrive at 12:05 when he should arrive before 11:54 if he was faster no ?

Reply:
The default speed limit in the infrastructure was 30 km/h, but with the code compo, the speed limit is set to 60 km/h. I updated the infrastructure speed limit to ensure the results are more consistent

Reply:
Oh didn't know about that. Thank you for changing it, it feels more "normal".